src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -162,7 +162,7 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
                             + " from " + ctx.getChannel());
                 }
                 NettyServerCnxn cnxn = (NettyServerCnxn)ctx.getAttachment();
-                synchronized(cnxn) {
+                synchronized(cnxn.getRpcLock()) {
                     processMessage(e, cnxn);
```

Comment:
> Now one thread can be in processMessage while another thread is getting stats about the connection. Is that ok?
>> I believe this is OK. We can receive two concurrent stat commands, but we only process one of them at a time. I'm also not a Netty wizard, so I could be wildly wrong :)
I'm not sure I agree with this. Since the results output by the stat command are no longer guaranteed to represent one single instance in time. 

Would it be possible to put a shared lock on the command's execution (shared with any other 4LW that acquires a lock on a connection other than its own, I THINK `cons` is the only other one)?